BrowseRouter/DefaultBrowserService.cs

```diff
@@ -8,18 +8,18 @@ public class DefaultBrowserService(INotifyService notifier)
   private const string AppName = "BrowseRouter";
   private const string AppID = "BrowseRouter";
   private const string AppDescription = "Opens a different brower based on the URL";
-  private string AppIcon => App.ExePath + ",0";
-  private string AppOpenUrlCommand => App.ExePath + " %1";
+  private static string AppIcon => App.ExePath + ",0";
```

Comment:
Please remove all of these static keywords. I understand your editor suggested these, but it does not know that just because something can be static does not mean it should be. 

Static is valuable where allocations are not wanted, such as in performance-critical code and hot paths. However, it's an enemy of maintainable code, because these members are not open for extension.

For example, keeping these nonstatic would keep open the possibility to expose them in an interface which `DefaultBrowserService` implements. Then they can be implemented in new ways, for example in test doubles. e.g. imagine `FakeBrowserService.GetRegisterStatus()`.

Reply:
I have trouble grasping how static could be an enemy of maintainable code, wouldn't it be revertable as soon as you need to make the variable instance-specific ?

Reply:
Oh, no because that would mean a potentially bigger refactoring needed as you'd need to change how the no-longer-static variable is accessed everywhere else in the code! that's what you meant ?

Reply:
Not necessarily if you've built other code that assumes it is static. The real problem here is static creates shared mutable state.

https://mb21.github.io/blog/2021/01/23/pure-functional-programming-and-shared-mutable-state

Static can be OK in the right context. This isn't it.

Reply:
Thanks! I understand!
And theoretically if this variable was a const (it can't be in this case due to App.ExePath), then it would be a shared UNmutable state and it wouldn't be so bad, right ? I suppose that's why the preceding lines are fine being 'const' even if it would also disrupt code evolutions relying on them being non-static.

Reply:
Yes, `const` makes it not mutable. So does `static readonly`, but `const` is special because it can be referenced at compile time.